### PR TITLE
OSAC-3160: Add GPU columns to InstanceType CLI table rendering

### PR DESCRIPTION
**Pull request by Tzif-Morgen**
## [OSAC-3160](https://redhat.atlassian.net/browse/OSAC-3160): Add GPU columns to InstanceType CLI table rendering

**Jira:** [OSAC-3160](https://redhat.atlassian.net/browse/OSAC-3160)
**Story type:** [DEV]

### Summary

Adds GPUS and GPU NAME columns to the InstanceType CLI table output (both public and private APIs), so users can identify which InstanceTypes include GPU hardware at a glance. Uses CEL `has()` guards to safely render blank values for non-GPU InstanceTypes.

### Changes

- **`internal/rendering/tables/osac.public.v1.InstanceType.yaml`** — Added GPUS, GPU NAME columns between MEMORY and STATE
- **`internal/rendering/tables/osac.private.v1.InstanceType.yaml`** — Same GPU columns for the private/admin API
- **`internal/rendering/table_renderer_test.go`** — Added "Optional GPU columns" test block verifying GPU-enabled and non-GPU InstanceType rendering; hoisted shared `renderInstanceTypes` helper

### Example Output
image

### Testing

- **Unit tests:** 2 new specs in "Optional GPU columns" block — GPU-enabled rendering and non-GPU blank columns
- **CEL compilation:** Existing "Compiles CEL expressions successfully for all table definitions" test covers nil-safety of `has()` guards for both public and private YAMLs
- **Integration tests:** Not affected — no API or server changes, table rendering only

### Acceptance Criteria

- [x] AC-1: InstanceType CLI table includes GPUS and GPU NAME columns
- [x] AC-2: GPU columns appear between MEMORY and STATE columns
- [x] AC-3: Non-GPU InstanceTypes show `0` / `-` in GPU columns
- [x] AC-4: GPU-enabled InstanceTypes show count and resource_name

## Summary by CodeRabbit

* **New Features**
  * Added GPU count and GPU resource name columns to instance type tables.
  * Displays configured GPU details for GPU-enabled instance types.
  * Shows `0` and `-` placeholders when GPU information is unavailable.

### CODE REVIEW

File: fulfillment-service/internal/rendering/table_renderer_test.go [hunk 1]
```diff
@@ -194,11 +194,61 @@ var _ = Describe("Table renderer", func() {
...
+							MemoryGib: 16,
+							State:     publicv1.InstanceTypeState_INSTANCE_TYPE_STATE_ACTIVE,
+						}.Build(),
+					}.Build(),
+				},
+			)
+			Expect(output).To(ContainSubstring("GPU DEVICE"))
+			Expect(output).To(MatchRegexp(`16\s+0\s+-\s+-\s+ACTIVE`))
+		})
+	})
```
[the remark this answers is not in the document]
Comment (Tzif-Morgen) [reply fulfillment-service/internal/rendering/table_renderer_test.go:249]: All existing rendering tests use `publicv1` only — the private table YAML is identical for the GPU columns (only the STATE enum type differs), and the CEL compilation test already covers both. Adding private-specific tests would duplicate coverage without catching new bugs. Dismissing as consistent with existing test patterns.

**Review by ygalblum — APPROVED** [review]
/approve
/lgtm
